backend/app/api/endpoints/internal/rag.py

```
@@ -252,6 +252,103 @@ async def get_knowledge_base_size(
+                message="Failed to update context record",
+            )
+
+    except Exception as e:
+        logger.error(
+            "[internal_rag] Save RAG result failed: subtask_id=%d, kb_id=%d, error=%s",
+            request.user_subtask_id,
+            request.knowledge_base_id,
+            e,
+            exc_info=True,
+        )
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)